src/lib/include/simd/simd_doubles.h

```diff
@@ -0,0 +1,162 @@
+#ifndef DISTOPIA_SIMD_DOUBLE_H
+#define DISTOPIA_SIMD_DOUBLE_H
+
+#include <cassert>
+#include <immintrin.h>
+
+// class to wrap __m128d 2 x packed doubles
+class SimdDoubleX2 {
+public:
+  inline void set(const double d);
+  inline void load(const double *source);
+  inline void store(double *target);
+  inline void loadU(const double *source);
+  inline void storeU(double *target);
+  inline void zero();
+  inline void reciprocal();
+
+  __m128d contents;
```

Comment:
The `foo.contents` syntax is really annoying. You can overload `=` to permit `foo = native_vector_bar` and a [custom conversion function](https://en.cppreference.com/w/cpp/language/cast_operator) so you can use this wrapper wherever you can use a native vector.